server/controllers/event.controller.js

```diff
@@ -1,17 +1,19 @@
 import Bluebird from 'bluebird';
-import producer from '../helpers/Producer';
+import EventStream from '../../api/EventStream';
+import Joi from 'joi';
+import paramValidation from '../../config/param-validation';
+import APIError from '../helpers/APIError';
 
-/*
-clientId: Joi.string().required(),
-      deviceType: Joi.string().required(),
-      location: Joi.string().required(),
-      eventTime: Joi.string().required(),
-      eventType: Joi.string().required()
-*/
+const producer = new EventStream.EventStreamProducer();
 
-function list(req, res, next) {
-	producer.showTopics((results) => {
-		res.json(JSON.stringify(results));
+function validateEventType(req, res, next) {
```

Comment:
@skabbass1 @crablar I'm adding validation for each event's `eventData` field. Been stuck on this for a little bit, but I keep getting a 500 error here. I couldn't see a way to easily add this validation into the [event.route](https://github.com/SoftwareEngineeringDaily/sedaily-event-stream/pull/18/files#diff-94e918abdbacc41e40bb8a9b36093d8cL16), so I added this function in the controller. Is there something obvious i'm missing here?

Reply:
@andrewmarklloyd  I dont fully understand your question - could you elaborate a bit or paste code snippet along with the text or  stack trace of the resulting 500 error here?  Thanks